dev-lang/ghc/ghc-9.4.2.ebuild

```
@@ -107,6 +107,15 @@ PREBUILT_BINARY_RDEPENDS="${PREBUILT_BINARY_DEPENDS}
 RDEPEND+="binary? ( ${PREBUILT_BINARY_RDEPENDS} )"
 
 DEPEND="${RDEPEND}"
+
+# see hadrian.cabal
+hadrian_deps="
+	dev-util/shake
+	dev-haskell/text
+	dev-haskell/unordered-containers
+	dev-haskell/transformers
+"
```

```suggestion
hadrian_deps="
	>=dev-haskell/cabal-3.2 <dev-haskell/cabal-3.7
	>=dev-util/shake-0.18.3  <dev-util/shake-0.20
	>=dev-haskell/unordered-containers-0.2.1 <dev-haskell/unordered-containers-0.3
"
```

---

- You don't need to include `dev-haskell/text` or `dev-haskell/transformers` since they come bundled.
- We should probably copy the version bounds from the `.cabal` file:
  - [`hadrian.cabal`](https://gitlab.haskell.org/ghc/ghc/-/blob/ghc-9.4.2-release/hadrian/hadrian.cabal#L150-162)
    ```cabal
        build-depends:       Cabal                >= 3.2     && < 3.7
                           , shake                >= 0.18.3  && < 0.20
                           , unordered-containers >= 0.2.1   && < 0.3
    ```
- Include `dev-haskell/cabal` since it has a relevant version restriction.

---

I notice that it depends on `Cabal <3.7`. How did this work considering `ghc-9.4.2` comes bundled with `Cabal-3.8.1.0`?

---

The dependency on shake may also be a problem since we will need to get around this dependency cycle.

---

I'm thinking we might want to split `hadrian` out into its own package with a mechanism similar to the `ghcbootstrap` USE flag, since it will require a functional `ghc` and `shake` to build it.